tests/dateSelector.spec.ts

```diff
@@ -0,0 +1,140 @@
+    //10. On the Owner Information page, add assertions for the newly created pet. Name is Tom, Birth Date is in the format "2014-05-02", Type is dog
+    const petField = page.locator('td', { has: page.getByText('Tom') })
+    await expect(petField).toContainText(`${birthYear}-${birthMonth}-${birthDay}`)
+    await expect(petField).toContainText('dog')
+    //11. Click "Delete Pet" button the for the new pet "Tom"
+    await page.getByRole('button', { name: 'Delete Pet' }).last().click()
+    //12. Add assertion that Tom does not exist in the list of pets anymore
+    await expect(petField).toHaveCount(0)
+})
+test('select the desired date in the calendar part II', async ({ page }) => {
+    // 2. In the list of the Owners, locate the owner by the name "Jean Coleman" and select this owner
+    await page.getByRole('link', { name: 'Jean Coleman' }).click()
+    //3. In the list of pets, locate the pet with a name "Samantha" and click "Add Visit" button
+    const addVisitButtonOwnerPage = page.locator('app-pet-list', { hasText: 'Samantha' })
+    await addVisitButtonOwnerPage.getByText('Add Visit').click()
+    //4. Add the assertion that "New Visit" is displayed as header of the page
+    await expect(page.getByRole("heading")).toHaveText("New Visit")
+    //5. Add the assertion that pet name is "Samantha" and owner name is "Jean Coleman"
+    await expect(page.locator('tr', { hasText: 'Samantha' })).toContainText('Jean Coleman')
+    //6. Click on the calendar icon and select the current date in date picker
+    const calendarIcon = page.getByLabel('Open calendar')
+    await calendarIcon.click()
+    await page.locator('span.mat-calendar-body-today').click()
+    //7. Add assertion that date is displayed in the format "YYYY/MM/DD"
+    const date = new Date()
+    const year = date.getFullYear();
+    const month = date.toLocaleString('en-US', { month: '2-digit' })
+    const day = date.toLocaleString('en-US', { day: '2-digit' })
+    const expectedDermatologyAppointmentDate = `${year}-${month}-${day}`
+    function formatDate(date: Date, expectedFormat: string): string {
+        if (!expectedFormat) {
+            throw new Error('The "expectedFormat" parameter is required. Please specify a valid date format.')
+        }
+        switch (expectedFormat) {
+            case 'YYYY/MM/DD':
+                return `${year}/${month}/${day}`
+            case 'YYYY-MM-DD':
+                return `${year}-${month}-${day}`
+            case 'MM YYYY':
+                return `${month} ${year}`;
+            default:
+                throw new Error('The "expectedFormat" parameter is required. Please specify a valid date format.')
+        }
+    }
+
+    const newVisitPageDateFormat = formatDate(date, 'YYYY/MM/DD')
+    const ownerPageDateFormat = formatDate(date, 'YYYY-MM-DD')
+
+    console.log('Formatted Date on New Visit Page:', newVisitPageDateFormat);
+    console.log('Formatted Date on Owner Page:', ownerPageDateFormat);
```

Comment:
Okay, I see a very good attempt at creating a function to format the date. Love it! Why did you abandon this approach and didn't use the function? :)

Reply:
Because it didn't work for me and I used AI for solve it. It was easier to keep it simple, agree.